examples/graph/graph-layer/graph-layout-layer.js

```diff
@@ -0,0 +1,148 @@
+import {CompositeLayer, COORDINATE_SYSTEM} from 'deck.gl';
+import GraphLayer, {GRAPH_LAYER_IDS} from './graph-layer';
+
+import LayoutD3 from './layout/layout-d3';
+
+const defaultProps = {
+  width: 640,
+  height: 640,
+  data: null,
+  opacity: 1.0,
+  layout: LayoutD3,
+  projectionMode: COORDINATE_SYSTEM.IDENTITY,
+  nodeIconAccessors: {}
+};
+
+/**
+ * GraphLayoutLayer displays a force-directed network graph in deck.gl.
+ * It accepts a list of nodes and a list of links,
+ * processes them with a graph layout engine (default: LayoutD3),
+ * and passes props and transformed data on to GraphLayer for rendering.
+ */
+export default class GraphLayoutLayer extends CompositeLayer {
+  initializeState() {
+    const {data, layoutAccessors} = this.props;
+    const Layout = this.props.layout;
+    this.state.layout = new Layout(Object.assign({data}, layoutAccessors));
+  }
+
+  updateState({oldProps, props, changeFlags}) {
+    const {layout} = this.state;
+    const data = props.data ? props.data[props.data.length - 1] : null;
+    let layoutData;
+
+    // If the data have changed, send to layout;
+    // else just update layout in its current state.
+    if (changeFlags.dataChanged) {
+      layoutData = data;
+    }
+
+    const {layoutProps} = this.props;
+    const {nodes, isUpdating} = layout.update(layoutData, layoutProps);
+    if (isUpdating) {
+      // update state only if layout is still running
+      this.state.nodes = nodes;
+      this.state.links = data ? data.links : undefined;
+      // create new object for updateTriggers diff
+      this.state.layoutUpdating = {};
```

Comment:
@gnavvy and I discussed; instead of using an Object, a timestamp is probably more legible to indicate that this is a value changing every update.